### PR TITLE
media-plugins/vdr-osdteletext: version bump to 2.3.1

### PR DESCRIPTION
**Pull request by madmartin**
Signed-off-by: Martin Dummer <martin.dummer@gmx.net>

### CODE REVIEW

File: media-plugins/vdr-osdteletext/vdr-osdteletext-2.3.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+SRC_URI="https://github.com/vdr-projects/vdr-plugin-osdteletext/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/vdr-plugin-osdteletext-${PV}"
+
+LICENSE="GPL-2+ public-domain" #teletext2.ttf, not copyrightable
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+BDEPEND="acct-user/vdr"
+DEPEND="media-video/vdr"
+RDEPEND="${DEPEND}"
```
Comment (juippis) [inline media-plugins/vdr-osdteletext/vdr-osdteletext-2.3.1.ebuild:19]:
```suggestion
RDEPEND="acct-group/vdr
	acct-user/vdr
	media-video/vdr"
DEPEND="${RDEPEND}"
```

**Review by juippis — COMMENTED** [review]
Can you spot what I'm doing wrong here? 
```
[INFO] >>> Running (1 of 1) =media-plugins/vdr-osdteletext-2.3.1 with USE: <default flags>

These are the packages that would be merged, in order:

Calculating dependencies  ... done!
[binary  N     ] media-fonts/corefonts-1-r7-1::gentoo  USE="-X -tahoma" 0 KiB
[ebuild  N     ] acct-group/vdr-0-r1::gentoo  0 KiB
[binary  N     ] acct-group/audio-0-r1-1::gentoo  0 KiB
[ebuild  N     ] acct-group/cdrom-0-r1::gentoo  0 KiB
[ebuild  N     ] acct-group/video-0-r1::gentoo  0 KiB
[binary  N     ] sys-process/wait_on_pid-0.0.2-r1-1::gentoo  0 KiB
[binary  N     ] media-libs/libjpeg-turbo-2.1.2-r1-1:0/0.2::gentoo  USE="-java -static-libs" ABI_X86="32 (64) (-x32)" 0 KiB
[binary  N     ] dev-util/pkgconf-1.8.0-r1-2:0/3::gentoo  USE="-test" ABI_X86="32 (64) (-x32)" 0 KiB
[ebuild  N     ] acct-user/vdr-0::gentoo  USE="-graphlcd -legacy-homedir -remote -serial -systemd" 0 KiB
[binary  N     ] media-libs/libpng-1.6.37-r2-1:0/16::gentoo  USE="-apng -static-libs" ABI_X86="32 (64) (-x32)" CPU_FLAGS_X86="sse" 0 KiB
[binary  N     ] virtual/ttf-fonts-1-r1-2::gentoo  0 KiB
[binary  N     ] virtual/jpeg-100-1::gentoo  USE="-static-libs" ABI_X86="32 (64) (-x32)" 0 KiB
[binary  N     ] virtual/pkgconfig-2-r1-2::gentoo  0 KiB
[binary  N     ] virtual/linuxtv-dvb-headers-5.8-1::gentoo  0 KiB
[binary  N     ] media-libs/freetype-2.11.1-2:2::gentoo  USE="adobe-cff bzip2 cleartype-hinting png -X -brotli -debug -doc -fontforge -harfbuzz (-infinality) -static-libs -utils" ABI_X86="32 (64) (-x32)" 0 KiB
[binary  N     ] app-eselect/eselect-vdr-0.0.4-1::gentoo  0 KiB
[binary  N     ] dev-lang/perl-5.34.0-r6-1:0/5.34::gentoo  USE="gdbm ithreads -berkdb -debug -doc -minimal" 0 KiB
[binary  N     ] virtual/perl-Data-Dumper-2.179.0-1::gentoo  0 KiB
[binary  N     ] virtual/perl-Test-Harness-3.430.0-1::gentoo  0 KiB
[binary  N     ] app-admin/perl-cleaner-2.30-1::gentoo  0 KiB
[binary  N     ] perl-core/File-Temp-0.231.100-3::gentoo  0 KiB
[binary  N     ] virtual/perl-File-Temp-0.231.100-3::gentoo  0 KiB
[binary  N     ] perl-core/Encode-3.120.0-3::gentoo  0 KiB
[binary  N     ] virtual/perl-Encode-3.120.0-3::gentoo  0 KiB
[binary  N     ] perl-core/CPAN-2.290.0-r1-3::gentoo  0 KiB
[binary  N     ] virtual/perl-CPAN-2.290.0-3::gentoo  0 KiB
[binary  N     ] media-tv/gentoo-vdr-scripts-2.8-1::gentoo  0 KiB
[binary  N     ] media-libs/fontconfig-2.13.1-r2-2:1.0::gentoo  USE="-doc -static-libs" ABI_X86="32 (64) (-x32)" 0 KiB
[binary  N     ] app-eselect/eselect-fontconfig-1.1-r1-2::gentoo  0 KiB
[binary  N     ] media-video/vdr-2.4.7-r1-1::gentoo  USE="-bidi -debug -demoplugins -html -keyboard -mainmenuhooks -menuorg -naludump -permashift -pinplugin -systemd -ttxtsubs -verbose" 0 KiB
[ebuild  N     ] media-plugins/vdr-osdteletext-2.3.1::gentoo  130 KiB

Total: 31 packages (31 new, 26 binaries), Size of downloads: 130 KiB

!!! The following binary packages have been ignored due to non matching USE:

    =sys-apps/file-5.41 -python_targets_python3_10 -python_targets_python3_8
    =sys-libs/libseccomp-2.5.3 -python_targets_python3_10 -python_targets_python3_8
    =media-fonts/dejavu-2.37 X
    =sys-apps/portage-3.0.30-r1 -python_targets_pypy3 -python_targets_python3_10 -python_targets_python3_8
    =sys-apps/pkgcore-0.12.9 -python_targets_python3_10 -python_targets_python3_8

NOTE: The --binpkg-respect-use=n option will prevent emerge
      from ignoring these binary packages if possible.
      Using --binpkg-respect-use=y will silence this warning.

>>> Verifying ebuild manifests
>>> Running pre-merge checks for acct-group/vdr-0-r1
>>> Running pre-merge checks for acct-group/audio-0-r1
>>> Running pre-merge checks for acct-group/cdrom-0-r1
>>> Running pre-merge checks for acct-group/video-0-r1
>>> Running pre-merge checks for acct-user/vdr-0
>>> Emerging binary (1 of 31) media-fonts/corefonts-1-r7::gentoo
>>> Emerging (2 of 31) acct-group/vdr-0-r1::gentoo
>>> Emerging binary (3 of 31) acct-group/audio-0-r1::gentoo
>>> Emerging (4 of 31) acct-group/cdrom-0-r1::gentoo
>>> Emerging (5 of 31) acct-group/video-0-r1::gentoo
>>> Emerging binary (6 of 31) sys-process/wait_on_pid-0.0.2-r1::gentoo
>>> Emerging binary (7 of 31) media-libs/libjpeg-turbo-2.1.2-r1::gentoo
>>> Emerging binary (8 of 31) dev-util/pkgconf-1.8.0-r1::gentoo
>>> Installing (7 of 31) media-libs/libjpeg-turbo-2.1.2-r1::gentoo
>>> Emerging binary (9 of 31) media-libs/libpng-1.6.37-r2::gentoo
>>> Installing (6 of 31) sys-process/wait_on_pid-0.0.2-r1::gentoo
>>> Emerging binary (10 of 31) virtual/linuxtv-dvb-headers-5.8::gentoo
>>> Installing (3 of 31) acct-group/audio-0-r1::gentoo
>>> Emerging binary (11 of 31) app-eselect/eselect-vdr-0.0.4::gentoo
>>> Installing (1 of 31) media-fonts/corefonts-1-r7::gentoo
>>> Emerging binary (12 of 31) virtual/jpeg-100::gentoo
>>> Installing (4 of 31) acct-group/cdrom-0-r1::gentoo
>>> Emerging binary (13 of 31) dev-lang/perl-5.34.0-r6::gentoo
>>> Installing (8 of 31) dev-util/pkgconf-1.8.0-r1::gentoo
>>> Emerging binary (14 of 31) media-tv/gentoo-vdr-scripts-2.8::gentoo
>>> Installing (5 of 31) acct-group/video-0-r1::gentoo
>>> Installing (9 of 31) media-libs/libpng-1.6.37-r2::gentoo
>>> Installing (10 of 31) virtual/linuxtv-dvb-headers-5.8::gentoo
>>> Installing (2 of 31) acct-group/vdr-0-r1::gentoo
>>> Emerging binary (15 of 31) virtual/pkgconfig-2-r1::gentoo
>>> Emerging binary (16 of 31) virtual/ttf-fonts-1-r1::gentoo
>>> Installing (11 of 31) app-eselect/eselect-vdr-0.0.4::gentoo
>>> Installing (12 of 31) virtual/jpeg-100::gentoo
>>> Emerging binary (17 of 31) media-libs/freetype-2.11.1::gentoo
>>> Emerging (18 of 31) acct-user/vdr-0::gentoo
>>> Installing (13 of 31) dev-lang/perl-5.34.0-r6::gentoo
>>> Installing (14 of 31) media-tv/gentoo-vdr-scripts-2.8::gentoo
>>> Emerging binary (19 of 31) virtual/perl-Data-Dumper-2.179.0::gentoo
>>> Installing (15 of 31) virtual/pkgconfig-2-r1::gentoo
>>> Installing (16 of 31) virtual/ttf-fonts-1-r1::gentoo
>>> Installing (17 of 31) media-libs/freetype-2.11.1::gentoo
>>> Installing (19 of 31) virtual/perl-Data-Dumper-2.179.0::gentoo
>>> Emerging binary (20 of 31) media-libs/fontconfig-2.13.1-r2::gentoo
>>> Emerging binary (21 of 31) virtual/perl-Test-Harness-3.430.0::gentoo
>>> Installing (20 of 31) media-libs/fontconfig-2.13.1-r2::gentoo
>>> Installing (21 of 31) virtual/perl-Test-Harness-3.430.0::gentoo
>>> Emerging binary (22 of 31) app-eselect/eselect-fontconfig-1.1-r1::gentoo
>>> Emerging binary (23 of 31) app-admin/perl-cleaner-2.30::gentoo
>>> Installing (18 of 31) acct-user/vdr-0::gentoo
>>> Installing (23 of 31) app-admin/perl-cleaner-2.30::gentoo
>>> Failed to install acct-user/vdr-0, Log file:
>>>  '/var/tmp/portage/acct-user/vdr-0/temp/build.log'
>>> Installing (22 of 31) app-eselect/eselect-fontconfig-1.1-r1::gentoo
>>> Jobs: 22 of 31 complete, 1 failed               Load avg: 0.38, 0.20, 0.08
 * Package:    acct-user/vdr-0
 * Repository: gentoo
 * Maintainer: martin.dummer@gmx.net proxy-maint@gentoo.org,vdr@gentoo.org
 * USE:        abi_x86_64 amd64 elibc_glibc kernel_linux userland_GNU
 * FEATURES:   network-sandbox preserve-libs sandbox userpriv usersandbox
>>> Unpacking source...
>>> Source unpacked in /var/tmp/portage/acct-user/vdr-0/work
>>> Preparing source in /var/tmp/portage/acct-user/vdr-0/work ...
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/acct-user/vdr-0/work ...
>>> Source configured.
>>> Compiling source in /var/tmp/portage/acct-user/vdr-0/work ...
>>> Source compiled.
>>> Test phase [not enabled]: acct-user/vdr-0

>>> Install acct-user/vdr-0 into /var/tmp/portage/acct-user/vdr-0/image
>>> Completed installing acct-user/vdr-0 into /var/tmp/portage/acct-user/vdr-0/image

 * Final size of build directory:  4 KiB
 * Final size of installed tree:  32 KiB

./
./usr/
./usr/lib/
./usr/lib/sysusers.d/
./usr/lib/sysusers.d/acct-user-vdr.conf
./var/
./var/lib/
./var/lib/vdr/
./var/lib/vdr/.keep_acct-user_vdr-0
>>> Done.
 * checking 2 files for package collisions
>>> Merging acct-user/vdr-0 to /
 * The user "vdr" exists on this system, his current home directory is "/var/vdr"
 * The new default home directory for user vdr is "/var/lib/vdr"
 * You have three options to continue:
 *  - set USE=legacy-homedir for acct-user/vdr to continue to use /var/vdr
 *  - move /var/vdr to /var/lib/vdr manually and repeat to install acct-user/vdr
 *  - move /var/vdr to /var/vdr.old or anywhere else (to keep it as your backup), repeat to
 *    install acct-user/vdr and let the installation create a fresh /var/lib/vdr
 * the emerge will stop here.
 * ERROR: acct-user/vdr-0::gentoo failed (preinst phase):
 *   user action required
 *
 * Call stack:
 *     ebuild.sh, line 127:  Called pkg_preinst
 *   environment, line 1225:  Called die
 * The specific snippet of code:
 *               die "user action required";
 *
 * If you need support, post the output of `emerge --info '=acct-user/vdr-0::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=acct-user/vdr-0::gentoo'`.
 * The complete build log is located at '/var/tmp/portage/acct-user/vdr-0/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/acct-user/vdr-0/temp/environment'.
 * Working directory: '/var/tmp/portage/acct-user/vdr-0/empty'
 * S: '/var/tmp/portage/acct-user/vdr-0/work'
!!! FAILED preinst: 1

 * Messages for package acct-group/vdr-0-r1:

 * Adding group 'vdr' to your system ...
 *  - Groupid: 452

 * Messages for package media-tv/gentoo-vdr-scripts-2.8:

 * Adding user 'vdr' to your system ...
 *  - Userid: 999
 *  - Shell: /sbin/nologin
 *  - Home: /var/vdr
 *  - Groups: vdr,video,audio,cdrom
 *  - GECOS: added by portage for gentoo-vdr-scripts
 *  - Creating /var/vdr in
 * media-tv/gentoo-vdr-scripts supports an init script
 * to start a X server
 * Please refer for detailed info to
 * /usr/share/doc/gentoo-vdr-scripts-2.8/README.x11-setup
 *
 * systemd is supported by media-tv/gentoo-vdr-scripts
 * This are described in the README.systemd file
 * in /usr/share/doc/gentoo-vdr-scripts-2.8/
 *
 * Plugins which should be used are set via
 * the config-file called /etc/conf.d/vdr.plugins
 * or enabled them via the frontend eselect vdr-plugin.
 *

 * Messages for package media-libs/fontconfig-2.13.1-r2:

 * Please make fontconfig configuration changes using `eselect
 * fontconfig`. Any changes made to /etc/fonts/fonts.conf will be
 * overwritten. If you need to reset your configuration to upstream
 * defaults, delete the directory /etc/fonts/conf.d/ and re-emerge
 * fontconfig.
 *
 * (Note: Above message is only printed the first time package is
 * installed. Please look at /usr/share/doc/fontconfig-2.13.1-r2/README.gentoo*
 * for future reference)

 * Messages for package acct-user/vdr-0:

 * The user "vdr" exists on this system, his current home directory is "/var/vdr"
 * The new default home directory for user vdr is "/var/lib/vdr"
 * You have three options to continue:
 *  - set USE=legacy-homedir for acct-user/vdr to continue to use /var/vdr
 *  - move /var/vdr to /var/lib/vdr manually and repeat to install acct-user/vdr
 *  - move /var/vdr to /var/vdr.old or anywhere else (to keep it as your backup), repeat to
 *    install acct-user/vdr and let the installation create a fresh /var/lib/vdr
 * the emerge will stop here.
 * ERROR: acct-user/vdr-0::gentoo failed (preinst phase):
 *   user action required
 *
 * Call stack:
 *     ebuild.sh, line 127:  Called pkg_preinst
 *   environment, line 1225:  Called die
 * The specific snippet of code:
 *               die "user action required";
 *
 * If you need support, post the output of `emerge --info '=acct-user/vdr-0::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=acct-user/vdr-0::gentoo'`.
 * The complete build log is located at '/var/tmp/portage/acct-user/vdr-0/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/acct-user/vdr-0/temp/environment'.
 * Working directory: '/var/tmp/portage/acct-user/vdr-0/empty'
 * S: '/var/tmp/portage/acct-user/vdr-0/work'

```

**Review by juippis — APPROVED** [review]
Tried emerging this without **any** binpkgs but still hit this error. What creates the legacy-homedir in that list?

**Comment by madmartin** [comment]
> What creates the legacy-homedir in that list?

Hello @juippis ,
USE=legacy-homedir is in acct-user/vdr 
In former times, media-tv/gentoo-vdr-scripts created user 'vdr' with HOME=/var/vdr which is not FHS conform.
acct-user/vdr wants to create HOME=/var/lib/vdr
Now with acct-user/vdr, to be backward compatible with existing installations (where /var/vdr can contain terabytes of video files.... like on my server) when acct-user/vdr is installed the first time, AND /var/vdr exists, then it stops. The user must decide: either set USE=legacy-homedir and keep on using /var/vdr - or rename/remove /var/vdr  and let portage create a new /var/lib/vdr on next installation of acct-user/vdr. 
Just to prevent a bad user experience.

**Comment by juippis** [comment]
Yes, thank you. That's what I meant, the container used here didn't exist this morning so it was a brand new clean system that hit that error. 

I'll keep an eye out for the new PR but there's a lot of backlog currently so no promises.

**Comment by madmartin** [comment]
@juippis media-tv/gentoo-vdr-scripts is ready in  #24059 if you have time for a review

**Pull request closed without merging**